aws_lambda_powertools/utilities/batch/middlewares.py

```diff
@@ -54,3 +56,58 @@ def batch_processor(
         processor.process()
 
     return handler(event, context)
+
+
+@lambda_handler_decorator
+def sqs_batch_processor(
+    handler: Callable,
+    event: Dict,
+    context: Dict,
+    record_handler: Callable,
+    config: Optional[Config] = None,
+    suppress_exception: bool = False,
+):
+    """
+    Middleware to handle SQS batch event processing
+
+    Parameters
+    ----------
+    handler: Callable
+        Lambda's handler
+    event: Dict
+        Lambda's Event
+    context: Dict
+        Lambda's Context
+    record_handler: Callable
+        Callable to process each record from the batch
+    config: Config
+            botocore config object
+    suppress_exception: bool, optional
+        Supress exception raised if any messages fail processing, by default False
+
+    Examples
+    --------
+    **Processes Lambda's event with PartialSQSProcessor**
+        >>> from aws_lambda_powertools.utilities.batch import sqs_batch_processor
+        >>>
+        >>> def record_handler(record):
+        >>>     return record["body"]
+        >>>
+        >>> @sqs_batch_processor(record_handler=record_handler)
+        >>> def handler(event, context):
+        >>>     return {"StatusCode": 200}
+
+    Limitations
+    -----------
+    * Async batch processors
+
+    """
+    config = config or Config()
+    processor = PartialSQSProcessor(config=config, suppress_exception=suppress_exception)
+
+    records = event["Records"]
+
+    with processor(records, record_handler):
+        processor.process()
```

Comment:
Can you add a `logger.debug` here to provide a message that you're processing another record. We don't need to log the record for security reasons, just a message suffices.

Reply:
There's no access to individual records here, but I've added logging elsewhere which should suffice.

Reply:
This is outside of the loop for processing single records, but I've added debug logging elsewhere which should suffice.